mlos_bench/mlos_bench/environments/local/local_env.py

```diff
@@ -63,13 +69,31 @@ def __init__(self,
         assert self._service is not None and isinstance(self._service, SupportsLocalExec), \
             "LocalEnv requires a service that supports local execution"
         self._local_exec_service: SupportsLocalExec = self._service
-
-        self._temp_dir = self.config.get("temp_dir")
+        self._temp_dir: Optional[str] = None
+        self._temp_dir_context: Union[TemporaryDirectory, nullcontext, None] = None
 
         self._dump_params_file: Optional[str] = self.config.get("dump_params_file")
         self._dump_meta_file: Optional[str] = self.config.get("dump_meta_file")
         self._read_results_file: Optional[str] = self.config.get("read_results_file")
 
+    def __enter__(self) -> Environment:
+        assert self._temp_dir is None and self._temp_dir_context is None
+        self._temp_dir_context = self._local_exec_service.temp_dir_context(self.config.get("temp_dir"))
```

Comment:
One hiccup with this could be that if `temp_dir` config is invalid for some reason (e.g., bad path passed in the config), then the error won't be propagated until runtime, rather than during initialization.  I think it's preferable to do the former whenever possible.

Perhaps we can add a quick sanity check on `config["temp_dir"]` during init that it is in fact a valid `Path` (or else `None`).

Reply:
I've added some checks in the constructor